### PR TITLE
vt v3 premium

### PR DESCRIPTION
**Pull request by jochman**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/20671

## Description
The VT-V3 premium integration. 
Please review code and yml. it still without outputs.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [x] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation 

## TODO:
- [ ] Generate outputs
- [ ] Unit-tests
- [ ] test-playbook

### CODE REVIEW

File: Packs/VirusTotal/Integrations/VirusTotal-V3-Premium-API/VirusTotal-V3-Premium-API.py [hunk 1]
```diff
@@ -0,0 +1,1001 @@
...
+    if not arg_to_boolean_can_be_none(args.get('extended_data')):
+        outputs = copy.deepcopy(raw_response)
+        outputs['data'] = decrease_data_size(outputs.get('data', []))
+    return CommandResults(
+        f'{INTEGRATION_ENTRY_CONTEXT}.LiveHuntFiles',
+        'id',
+        readable_output=tableToMarkdown(
+            'Notifications file listed:',
+            [
+                {**item.get('attributes', {}), 'id': item.get('id')} for item in raw_response.get('data', [])
```
Comment (Itay4) [inline Packs/VirusTotal/Integrations/VirusTotal-V3-Premium-API/VirusTotal-V3-Premium-API.py:782]: please add `headers`

**Review by Itay4 — COMMENTED** [review]
looks good 👍 
partial review
1. i see some apis have this:
image
please make sure it is documented in the integration readme
2. for every command argument (query or filter), please add an example and refer to the vt docs for the syntax
3. like in the public api, im concerned with the size of data we output to the context, lets have another look at whats actually important to be there

File: Packs/VirusTotal/Integrations/VirusTotal-V3-Premium-API/VirusTotal-V3-Premium-API.py [hunk 2]
```diff
@@ -0,0 +1,1001 @@
...
+    """Retrieve information about a ZIP file"""
+    zip_id = args['zip_id']
+    raw_response = client.get_zip(zip_id)
+    data = raw_response['data']
+    status = data.get('attributes', {}).get('status')
+    return CommandResults(
+        f'{INTEGRATION_ENTRY_CONTEXT}.Zip',
+        'id',
+        outputs=data,
+        readable_output=f'Zipping status is "{status}"!',
```
Comment (Itay4) [inline Packs/VirusTotal/Integrations/VirusTotal-V3-Premium-API/VirusTotal-V3-Premium-API.py:598]:
```suggestion
        readable_output=f'ZIP creation status is "{status}"',
```

File: Packs/VirusTotal/Integrations/VirusTotal-V3-Premium-API/VirusTotal-V3-Premium-API.py [hunk 3]
```diff
@@ -0,0 +1,1001 @@
...
+        )
+    )
+
+
+def delete_livehunt_rules(client: Client, args: dict) -> CommandResults:
+    """Delete a VT Hunting Livehunt ruleset."""
+    id_ = args['id']
+    client.delete_livehunt_rule(id_)
+    return CommandResults(
+        readable_output=f'Rule ID: "{id_}" has been successfully deleted.'
```
Comment (Itay4) [inline Packs/VirusTotal/Integrations/VirusTotal-V3-Premium-API/VirusTotal-V3-Premium-API.py:738]:
```suggestion
        readable_output=f'Rule "{id_}" was deleted successfully'
```

File: Packs/VirusTotal/Integrations/VirusTotal-V3-Premium-API/VirusTotal-V3-Premium-API.yml [hunk 4]
```diff
@@ -0,0 +1,531 @@
...
+      isArray: false
+      name: enabled
+      required: false
+      secret: false
+      defaultValue: "None"
+      auto: PREDEFINED
+      predefined:
+      - "true"
+      - "false"
+      - "None"
```
Comment (Itay4) [inline Packs/VirusTotal/Integrations/VirusTotal-V3-Premium-API/VirusTotal-V3-Premium-API.yml:233]: lets remove the `None` option and just keep this argument without default value

File: Packs/VirusTotal/Integrations/VirusTotal-V3-Premium-API/VirusTotal-V3-Premium-API.yml [hunk 5]
```diff
@@ -0,0 +1,531 @@
...
+    name: vt-premium-intelligence-search
+  - arguments:
+    - default: true
+      description: Ruleset identifier.
+      isArray: false
+      name: id
+      required: true
+      secret: false
+    deprecated: false
+    description: Retrieve VT Hunting livehunt rulesets.
```
Comment (Itay4) [inline Packs/VirusTotal/Integrations/VirusTotal-V3-Premium-API/VirusTotal-V3-Premium-API.yml:185]:
```suggestion
    description: Retrieve VT Hunting livehunt ruleset.
```

File: Packs/VirusTotal/Integrations/VirusTotal-V3-Premium-API/VirusTotal-V3-Premium-API.py [hunk 6]
```diff
@@ -0,0 +1,1001 @@
...
+    readable_output = [item.get('attributes', {}) for item in data]
+    return CommandResults(
+        f'{INTEGRATION_ENTRY_CONTEXT}.LiveHuntRule',
+        'id',
+        outputs=data,
+        raw_response=raw_response,
+        readable_output=tableToMarkdown(
+            'Presented rules:',
+            readable_output,
+            headers=['name', 'enabled', 'rule_names']
```
Comment (Itay4) [inline Packs/VirusTotal/Integrations/VirusTotal-V3-Premium-API/VirusTotal-V3-Premium-API.py:677]: please add the id to the human readable output

File: Packs/VirusTotal/Integrations/VirusTotal-V3-Premium-API/VirusTotal-V3-Premium-API.py [hunk 7]
```diff
@@ -0,0 +1,1001 @@
...
+    )
+    data = raw_response.get('data', [])
+    readable_output = [item.get('attributes', {}) for item in data]
+    return CommandResults(
+        f'{INTEGRATION_ENTRY_CONTEXT}.LiveHuntRule',
+        'id',
+        outputs=data,
+        raw_response=raw_response,
+        readable_output=tableToMarkdown(
+            'Presented rules:',
```
Comment (Itay4) [inline Packs/VirusTotal/Integrations/VirusTotal-V3-Premium-API/VirusTotal-V3-Premium-API.py:675]:
```suggestion
            'VT Hunting Livehunt rulesets',
```

File: Packs/VirusTotal/Integrations/VirusTotal-V3-Premium-API/VirusTotal-V3-Premium-API.py [hunk 8]
```diff
@@ -0,0 +1,1001 @@
...
+    raw_response = client.get_livehunt_rule_by_id(id_)
+    data = raw_response.get('data', {})
+    readable_output = data.get('attributes')
+    return CommandResults(
+        f'{INTEGRATION_ENTRY_CONTEXT}.LiveHuntRule',
+        'id',
+        outputs=data,
+        raw_response=raw_response,
+        readable_output=tableToMarkdown(
+            'Presented rules:',
```
Comment (Itay4) [inline Packs/VirusTotal/Integrations/VirusTotal-V3-Premium-API/VirusTotal-V3-Premium-API.py:650]:
```suggestion
            f'Livehunt Ruleset {id_}',
```

File: Packs/VirusTotal/Integrations/VirusTotal-V3-Premium-API/VirusTotal-V3-Premium-API.py [hunk 9]
```diff
@@ -0,0 +1,1001 @@
...
+            try:
+                del data['attributes'][attribute]
+            except KeyError:
+                pass
+    return data
+
+
+def raise_if_hash_not_valid(
+        file_hash: str,
+        valid_hashes: Union[tuple, str] = ('sha256', 'sha1', 'md5')
```
[the remark this answers is not in the document]
Comment (jochman) [reply Packs/VirusTotal/Integrations/VirusTotal-V3-Premium-API/VirusTotal-V3-Premium-API.py:142]: It's only for internal use. That's the output of `get_hash_type`

File: Packs/VirusTotal/Integrations/VirusTotal-V3-Premium-API/VirusTotal-V3-Premium-API.py [hunk 10]
```diff
@@ -0,0 +1,1001 @@
...
+    content = response.content
+    file_name = get_file_name(response.headers.get('Content-Disposition', ''))
+    return fileResult(file_name, content)
+
+
+def get_pcap_behaviour(client: Client, args: dict) -> dict:
+    """Extracted PCAP from a sandbox analysis"""
+    report_id = args['report_id']
+    content = client.get_pcap_beaviour(report_id)
+    assert isinstance(content, (bytes, str)), 'Response from PCAP Behavior is not a bytes-like object.'
```
[the remark this answers is not in the document]
Comment (jochman) [reply Packs/VirusTotal/Integrations/VirusTotal-V3-Premium-API/VirusTotal-V3-Premium-API.py:616]: It's for mypy.

File: Packs/VirusTotal/Integrations/VirusTotal-V3-Premium-API/VirusTotal-V3-Premium-API.yml [hunk 11]
```diff
@@ -0,0 +1,531 @@
...
+      secret: false
+      defaultValue: "None"
+      auto: PREDEFINED
+      predefined:
+      - "true"
+      - "false"
+      - "None"
+    description: Retrieve VT Hunting livehunt rulesets.
+    execution: false
+    name: vt-premium-livehunt-rules-list
```
[the remark this answers is not in the document]
Comment (jochman) [reply Packs/VirusTotal/Integrations/VirusTotal-V3-Premium-API/VirusTotal-V3-Premium-API.yml:236]: Well, yeah 
image

File: Packs/VirusTotal/Integrations/VirusTotal-V3-Premium-API/VirusTotal-V3-Premium-API.yml (hunk 5 above)
Comment (jochman) [reply Packs/VirusTotal/Integrations/VirusTotal-V3-Premium-API/VirusTotal-V3-Premium-API.yml:185]: It's rulests, not ruleset. And it's a copy-paste from vt description

File: Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/VirusTotal_V3_Premium.yml [hunk 12]
```diff
@@ -0,0 +1,1545 @@
...
+    - contextPath: VirusTotal.RetroHuntJobFiles.context_attributes.match_in_subfile
+      description: whether the match was in a subfile or not.
+      type: Boolean
+    deprecated: false
+    description: Retrieve matches for a retrohunt job matching file relationship.
+    execution: false
+    name: vt-premium-retrohunt-jobs-get-matching-files
+  - arguments:
+    - default: false
+      description: The rules to create.
```
Comment (Itay4) [inline Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/VirusTotal_V3_Premium.yml:1265]: please elaborate here.. add example and link to doc

File: Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/VirusTotal_V3_Premium.yml [hunk 13]
```diff
@@ -0,0 +1,1547 @@
...
+    - contextPath: VirusTotal.LiveHuntRule.links.self
+      description: Link to the ruleset
+      type: String
+    deprecated: false
+    description: Update a VT Hunting Livehunt ruleset.
+    execution: false
+    name: vt-premium-livehunt-rules-update
+  - arguments:
+    - default: true
+      description: Ruleset identifier
```
Comment (Itay4) [inline Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/VirusTotal_V3_Premium.yml:622]: add how to retrieve

File: Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/VirusTotal_V3_Premium.yml [hunk 14]
```diff
@@ -0,0 +1,1547 @@
...
+    - contextPath: VirusTotal.RetroHuntJob.links.self
+      description: Link to the entry
+      type: String
+    deprecated: false
+    description: Get a VT Hunting Livehunt ruleset by hunting notification files relationship.
+    execution: false
+    name: vt-premium-retrohunt-jobs-list
+  - arguments:
+    - default: true
+      description: Job identifier
```
Comment (Itay4) [inline Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/VirusTotal_V3_Premium.yml:1014]: add how to retrieve

File: Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/VirusTotal_V3_Premium.yml [hunk 15]
```diff
@@ -0,0 +1,1547 @@
...
+    - contextPath: VirusTotal.RetroHuntJob.links.self
+      description: Link to the entry
+      type: String
+    deprecated: false
+    description: Retrieve a retrohunt job.
+    execution: false
+    name: vt-premium-retrohunt-jobs-get-by-id
+  - arguments:
+    - default: true
+      description: Job identifier
```
Comment (Itay4) [inline Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/VirusTotal_V3_Premium.yml:1071]: add how to retrieve

File: Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/VirusTotal_V3_Premium.py [hunk 16]
```diff
@@ -0,0 +1,1028 @@
...
+    corpus = args.get('corpus')  # main / goodware
+    start_time, end_time = args.get('start_time'), args.get('end_time')
+    time_range = get_time_range_object(start_time, end_time)
+    raw_response = client.create_retrohunt_job(rules, notification_email, corpus, time_range)
+    data = raw_response.get('data', {})
+    return CommandResults(
+        f'{INTEGRATION_ENTRY_CONTEXT}.RetroHuntJob',
+        'id',
+        readable_output=tableToMarkdown(
+            'Retrohunt job has been successfully created!',
```
Comment (Itay4) [inline Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/VirusTotal_V3_Premium.py:863]:
```suggestion
            'Retrohunt job has been successfully created!,
```

File: Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/VirusTotal_V3_Premium.py [hunk 17]
```diff
@@ -0,0 +1,1028 @@
...
+def get_retrohunt_job_by_id(client: Client, args: dict) -> CommandResults:
+    """Retrieve a retrohunt job."""
+    id_ = args['id']
+    raw_response = client.get_retrohunt_job_by_id(id_)
+    data = raw_response.get('data', {})
+    return CommandResults(
+        f'{INTEGRATION_ENTRY_CONTEXT}.RetroHuntJob',
+        'id',
+        readable_output=tableToMarkdown(
+            'Retrohunt job has been successfully created!',
```
Comment (Itay4) [inline Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/VirusTotal_V3_Premium.py:884]:
```suggestion
            f'Retrohunt job {id_}',
```

File: Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/VirusTotal_V3_Premium.py [hunk 18]
```diff
@@ -0,0 +1,1028 @@
...
+                **data.get('attributes', {})
+            },
+            headers=['id', 'corpus', 'status', 'rules']
+        ),
+        outputs=data,
+        raw_response=raw_response
+    )
+
+
+def get_retrohunt_job_by_id(client: Client, args: dict) -> CommandResults:
```
Comment (Itay4) [inline Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/VirusTotal_V3_Premium.py:875]: please convert timestamps to human readable dates
image

File: Packs/VirusTotal/Integrations/VirusTotal-V3-Premium-API/VirusTotal-V3-Premium-API.yml [hunk 19]
```diff
@@ -0,0 +1,531 @@
...
+      name: yara_rule
+      required: false
+      secret: false
+    - default: false
+      description: Whatever to enable the rule.
+      isArray: false
+      name: enabled
+      required: false
+      secret: false
+      defaultValue: "false"
```
[the remark this answers is not in the document]
Comment (jochman) [reply Packs/VirusTotal/Integrations/VirusTotal-V3-Premium-API/VirusTotal-V3-Premium-API.yml]: well, it'll be disabled anyway if not supplied. So "false" default value will be much more clear to the user.

**Comment by ShirleyDenkberg** [comment]
@Itay4 Submitted. In order to meet the requirements of completing all PRs within a week, there was not enough time to review the descriptions of the arguments. If required, we can revisit this at a later time.

File: Packs/VirusTotal/Classifiers/classifier-VirusTotal_Intelligence_LiveHunt_Notification.json [hunk 20]
```diff
@@ -0,0 +1,66 @@
+{
```
Comment (Itay4) [inline Packs/VirusTotal/Classifiers/classifier-VirusTotal_Intelligence_LiveHunt_Notification.json:1]: please add classifier from 5.5

File: Packs/VirusTotal/IncidentFields/rulename.json [hunk 21]
```diff
@@ -0,0 +1,28 @@
+{
```
Comment (Itay4) [inline Packs/VirusTotal/IncidentFields/rulename.json:1]: please consult with @ytauman and @Arsenikr if the incident fields should be in common types

File: Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/README.md [hunk 22]
```diff
@@ -0,0 +1,1992 @@
...
+- Get a PCAP file generated from VirusTotal's sandbox for further analysis. 
+
+## Configure VirusTotal - Premium (API v3) on Cortex XSOAR  
+1. Navigate to **Settings** > **Integrations** > **Servers & Services**.
+2. Search for VirusTotal - Premium (API v3).
+3. Click **Add instance** to create and configure a new integration instance.
+
+    | **Parameter** | **Required** |
+    | --- | --- |
+    |  | True |
```
Comment (Itay4) [inline Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/README.md:20]: please fix

File: Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/README.md [hunk 23]
```diff
@@ -0,0 +1,1992 @@
...
+
+## Fetch Incidents
+Fetch incidents will fetch livehunt notifications of the given ruleset or identifier. 
+As an example, you can fetch incidents that were created by the CyberGate ruleset by using the "cybergate" tag.
+
+## VirusTotal - Private API compatibility
+The following 2 commands appear in both the *VirusTotal - Private API* and *VirusTotal Premium - (API v3)* integrations.
+- **vt-private-search-file**
+- **vt-private-download-file**
+ It is recommended to use these commands in the *VirusTotal Premium - (API v3)* integration. For all other commands, you should use the *VirusTotal - Private API* integration.
```
Comment (Itay4) [inline Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/README.md:44]:
```suggestion
 It is recommended to use these commands in the *VirusTotal Premium - (API v3)* integration. 
 
 For all other commands, you should use the *VirusTotal - Private API* integration.
```

File: Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/README.md [hunk 24]
```diff
@@ -0,0 +1,1992 @@
...
+| File.Size | Number | The size of the file in bytes. | 
+| File.MD5 | String | The MD5 hash of the file. | 
+| File.SHA1 | String | The SHA1 hash of the file. | 
+| File.SHA256 | String | The SHA256 hash of the file. | 
+| File.SHA512 | String | The SHA512 hash of the file. | 
+| File.Extension | String | The file extension. For example: "xls". | 
+| File.Type | String | The file type, as determined by libmagic \(same as displayed in file entries\). | 
+
+
+#### Command Example
```
Comment (Itay4) [inline Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/README.md:256]: add example

File: Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/README.md [hunk 25]
```diff
@@ -0,0 +1,1992 @@
...
+``` ```
+
+#### Human Readable Output
+
```
Comment (Itay4) [inline Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/README.md:260]: remove empty outputs

File: Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/README.md [hunk 26]
```diff
@@ -0,0 +1,1992 @@
...
+4. Click **Test** to validate the URLs, token, and connection.
+
+### Acquire API Key 
+Your API key can be found in your VirusTotal account user menu:  
+![](https://files.readme.io/ddeb298-Screen_Shot_2019-10-17_at_3.17.04_PM.png)  
+Your API key carries all your privileges, so keep it secure and don't share it with anyone.
+
+## Fetch Incidents
+Fetch incidents will fetch livehunt notifications of the given ruleset or identifier.
```
Comment (Itay4) [inline Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/README.md:37]: please add note here about the scope of the ruleset

File: Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/VirusTotal_V3_Premium.yml (hunk 12 above)
Comment (Itay4) [reply Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/VirusTotal_V3_Premium.yml:1507]: add link to doc

File: Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/VirusTotal_V3_Premium_description.md [hunk 27]
```diff
@@ -0,0 +1,13 @@
+## VirusTotal - Premium API - V3
+Analyze retro hunts, read live hunt notification,s and download files from VirusTotal.
+The premium API is a component of VirusTotal's [advanced services for professionals](https://www.virustotal.com/gui/services-overview).
+
+## Authorization:
+Your API key can be found in your VirusTotal account user menu:  
+![](https://files.readme.io/ddeb298-Screen_Shot_2019-10-17_at_3.17.04_PM.png)
```
Comment (Itay4) [inline Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/VirusTotal_V3_Premium_description.md:7]: remove image

File: Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/VirusTotal_V3_Premium_description.md [hunk 28]
```diff
@@ -0,0 +1,13 @@
+## VirusTotal - Premium API - V3
+Analyze retro hunts, read live hunt notification,s and download files from VirusTotal.
+The premium API is a component of VirusTotal's [advanced services for professionals](https://www.virustotal.com/gui/services-overview).
+
+## Authorization:
+Your API key can be found in your VirusTotal account user menu:  
+![](https://files.readme.io/ddeb298-Screen_Shot_2019-10-17_at_3.17.04_PM.png)  
+Your API key carries all your privileges, so keep it secure and don't share it with anyone.
+
```
Comment (Itay4) [inline Packs/VirusTotal/Integrations/VirusTotal_V3_Premium/VirusTotal_V3_Premium_description.md:9]: add note about what should be filled in the username for before 6.1

**Review by Itay4 — APPROVED** [review]
please address the comment https://github.com/demisto/content/pull/11825/files#r608580518

**Pull request merged**